provider/lxd/environ.go

```diff
@@ -156,15 +156,11 @@ func (env *environ) SetCloudSpec(_ stdcontext.Context, spec environscloudspec.Cl
 	defer env.lock.Unlock()
 
 	serverFactory := env.provider.serverFactory
-	server, err := serverFactory.RemoteServer(spec)
+	server, err := serverFactory.RemoteServer(spec, env.ecfgUnlocked.project())
```

Comment:
I'm not a fan of this, I think at the very least we should wrap the spec to become an `LXDCloudSpec` with a `Project` field.